hermes/crates/cbork/cddl-parser/src/grammar/rfc_8610.pest

```diff
@@ -103,58 +105,33 @@ int = ${ "-"? ~ uint }
 
 /// Unsigned Integers
 uint = ${
-    ( ASCII_NONZERO_DIGIT ~ ASCII_DIGIT* )
-    | ( "0x" ~ ASCII_HEX_DIGIT+ )
-    | ( "0b" ~ ASCII_BIN_DIGIT+ )
+    (ASCII_NONZERO_DIGIT ~ ASCII_DIGIT*)
+    | ("0x" ~ ASCII_HEX_DIGIT+)
+    | ("0b" ~ ASCII_BIN_DIGIT+)
     | "0"
 }
 
 /// Literal Text
 text = ${ "\"" ~ SCHAR* ~ "\"" }
 
-/// Literal Bytes - Note CDDL Spec incorrectly defines b64''.
-bytes = ${ bytes_hex | bytes_b64 | bytes_text }
-bytes_hex = ${ "h" ~ "'" ~ HEX_PAIR* ~ "'" }
-bytes_b64 = ${ "b64" ~ "'" ~ URL_BASE64 ~ "'" }
-bytes_text = ${ "'" ~ BCHAR* ~ "'" }
+/// Literal Bytes.
```

Comment:
It has parsing ambiguity there. For instance:

Parsing `gen-value = ( b64'asdf': uint )` produces pairs like this:

```
- grpent > group > grpchoice > grpent
   - memberkey > value > bytes > bytes_b64: "b64'asdf'"
```

While parsing `gen-value = ( h'atat': uint )` produces pairs like this:

```
- grpent > group > grpchoice
   - grpent > type > type1 > type2 > typename > id > name: "h"
   - optcom: ""
   - grpent
    - memberkey > value > bytes > bytes_text: "'atat'"
```

instead of:

```
- grpent > group > grpchoice > grpent
   - memberkey > value > bytes > bytes_hex: "h'atat'"
```

So this is unexpected as it neither detects the hex pairs error nor throws a parsing error.

I generalized `bytes` by just accepting `BCHAR`. There will be no hex pairs or b64 checks as these will be the task for semantic checks for the validity of the value inside `bytes`. Also from the CDDL spec, it doesn't have variants of `bytes` rules in their ABNF. So it's better to stick around with theirs (even though some rules might be incorrect).

Reply:
We need to work out why there is a parsing ambiguity, because `bytes` was testing properly.
Check and see if there is a parsing ambiguity directly on the `bytes` RULE.